### PR TITLE
DAOS-12985 test: nvme/fault.py 'dict' object has no attribute 'stdou_text"

### PR DESCRIPTION
**Pull request by dinghwah**
Description:
Update nvme_utils.py set_device_faulty result to parse json output.

Test-tag: test_nvme_fault
Test-repeat: 3
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/nvme_utils.py [hunk 1]
```diff
@@ -208,7 +208,8 @@ def set_device_faulty(self, server, disk_id):
         self.dmg.storage_set_faulty(disk_id)
         result = self.dmg.storage_query_device_health(disk_id)
         # Check if device state changed to EVICTED.
-        if 'State:EVICTED' not in result.stdout_text:
+        if result['response']['host_storage_map']['storage']['smd_info']['devices']['dev_state'] \
+                != 'EVICTED':
```
Comment (daltonbohning) [inline src/tests/ftest/util/nvme_utils.py:212]: I think it's recommended to do something like this:
https://github.com/daos-stack/daos/blob/dc0beb982a9be6f190f0819c25c5655a044cf9ff/src/tests/ftest/nvme/health.py#L115

@phender Would know better

Comment (phender) [inline src/tests/ftest/util/nvme_utils.py:212]: Use code similar to https://github.com/daos-stack/daos/blob/master/src/tests/ftest/nvme/health.py#L115-L133
```suggestion
        info = get_dmg_smd_info(self, dmg.storage_set_faulty, 'devices', uuid=uuid)
        for devices in info.values():
            for device in devices:
                if device['uuid'] != uuid:
                    continue
                if device['dev_state'].lower() != 'evicted':
                    self.fail("device State {} on host {} suppose to be EVICTED".format(disk_id, server))
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/nvme_utils.py (hunk 1 above)
Comment (dinghwah) [reply src/tests/ftest/util/nvme_utils.py]: Please see the update.

File: src/tests/ftest/util/nvme_utils.py [hunk 2]
```diff
@@ -205,11 +205,15 @@ def set_device_faulty(self, server, disk_id):
...
-            self.fail("device State {} on host {} suppose to be EVICTED".format(disk_id, server))
+        info = get_dmg_smd_info(
+            self, self.dmg.storage_set_faulty(disk_id), 'devices', uuid=disk_id)
+        for devices in info.values():
+            for device in devices:
+                if device['uuid'] != disk_id:
+                    continue
+                if device['dev_state'].lower() != 'evicted':
+                    self.fail(
+                        "device {} State on host {} suppose to be EVICTED".format(disk_id, server))
```
Comment (shimizukko) [inline src/tests/ftest/util/nvme_utils.py:216]:
```suggestion
                        "State of device {} on host {} is suppose to be EVICTED".format(disk_id, server))
```

**Review by shimizukko — DISMISSED** [review]

**Review by shimizukko — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/util/nvme_utils.py [hunk 3]
```diff
@@ -205,11 +205,15 @@ def set_device_faulty(self, server, disk_id):
             disk_id (string): NVMe disk ID where it will be changed to faulty.
         """
         self.dmg.hostlist = server
-        self.dmg.storage_set_faulty(disk_id)
-        result = self.dmg.storage_query_device_health(disk_id)
-        # Check if device state changed to EVICTED.
-        if 'State:EVICTED' not in result.stdout_text:
-            self.fail("device State {} on host {} suppose to be EVICTED".format(disk_id, server))
+        info = get_dmg_smd_info(
+            self, self.dmg.storage_set_faulty, 'devices', uuid=disk_id)
```
Comment (phender) [inline src/tests/ftest/util/nvme_utils.py:209]:
```suggestion
        self.dmg.storage_set_faulty(disk_id)
        info = get_dmg_smd_info(
            self, self.dmg.storage_query_device_health, 'devices', uuid=disk_id)
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/nvme_utils.py (hunk 3 above)
Comment (dinghwah) [reply src/tests/ftest/util/nvme_utils.py]: Please see the update.

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**